### PR TITLE
JAMES-2669 Add Aws S3 BlobStore configuration

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/AwsS3ConfigurationReader.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.net.URI;
+
+import org.apache.commons.configuration.Configuration;
+
+import com.google.common.base.Strings;
+import com.google.common.base.Preconditions;
+
+public class AwsS3ConfigurationReader {
+
+    static final String OBJECTSTORAGE_ACCESKEYID =
```
Comment (mbaechler) [inline server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/AwsS3ConfigurationReader.java]: please don't duplicate `objectstorage` prefix as it's used in different classes (far from each others)

File: server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageBlobConfiguration.java [hunk 2]
```diff
@@ -143,6 +128,59 @@ public ReadyToBuild keystone3(SwiftKeystone3ObjectStorage.Configuration authConf
...
+
+                    private final String providerName;
+                    private final Factory factory;
+
+                    private Factories(String providerName, Factory factory) {
+                        this.providerName = providerName;
+                        this.factory = factory;
+                    }
+
+                    public static Optional<Factory> find(String providerName) {
```
Comment (mbaechler) [inline server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageBlobConfiguration.java]: I would put this method into `Builder` and name it `forProvider`

File: server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageBlobConfiguration.java [hunk 3]
```diff
@@ -143,6 +128,59 @@ public ReadyToBuild keystone3(SwiftKeystone3ObjectStorage.Configuration authConf
...
+                        return Stream.of(values())
+                            .filter(factory -> factory.providerName.equalsIgnoreCase(providerName))
```
Comment (mbaechler) [inline server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageBlobConfiguration.java]: we usually care about case sensitiveness

File: server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageBlobConfiguration.java [hunk 4]
```diff
@@ -143,6 +128,59 @@ public ReadyToBuild keystone3(SwiftKeystone3ObjectStorage.Configuration authConf
 
         public static class ReadyToBuild {
 
+            public static interface Factory {
+                static enum Factories {
+                    SWIFT("swift", new Swift()),
```
Comment (mbaechler) [inline server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageBlobConfiguration.java]: you don't really need an instance (you don't need a class, neither), Factory being a functional interface, you can implement it with a single method.

**Comment by Arsnael** [comment]
Read it...

**Comment by aduprat** [comment]
Replaced by https://github.com/linagora/james-project/pull/2198

**Pull request closed without merging**